File: backend/app/schemas/qa.py
from typing import Literal

from pydantic import BaseModel, Field


class Citation(BaseModel):
    meeting_id: str
    chunk_id: str
    speaker: str
    start: float = Field(ge=0)
    end: float = Field(ge=0)
    text: str


class QARequest(BaseModel):
    meeting_id: str | None = None
    question: str = Field(min_length=1)
    scope: Literal["current_meeting", "all_meetings"] = "current_meeting"


class QAResponse(BaseModel):
    answer: str
    citations: list[Citation]


class WSMessage(BaseModel):
    type: Literal["start", "chunk", "end", "error", "job_progress"]
    message: str | None = ""
    job_id: str | None = None
    progress: int | None = Field(default=None, ge=0, le=100)
    citations: list[Citation] | None = None

File: frontend/meetingPilot-frontend/src/types/qa.ts
export interface Citation {
  meeting_id: string
  chunk_id: string
  speaker: string
  start: number
  end: number
  text: string
}

export interface QARequest {
  meeting_id?: string | null
  question: string
  scope: 'current_meeting' | 'all_meetings'
}

export interface QAResponse {
  answer: string
  citations: Citation[]
}

export type WSMessageType = 'start' | 'chunk' | 'end' | 'error' | 'job_progress'

export interface WSMessage {
  type: WSMessageType
  message?: string | null
  job_id?: string | null
  progress?: number | null
  citations?: Citation[] | null
}

export interface MemorySearchResultItem {
  meeting_id: string
  meeting_title: string
  chunk_id: string
  speaker: string
  start: number
  end: number
  text: string
  score: number
}

export interface MemorySearchResponse {
  items: MemorySearchResultItem[]
}

export interface GraphNode {
  id: string
  label: string
  type: string
}

export interface GraphEdge {
  source: string
  target: string
  type: string
}

export interface MeetingGraphResponse {
  nodes: GraphNode[]
  edges: GraphEdge[]
}

export interface GraphQueryResponse {
  answer: string
  nodes: GraphNode[]
  edges: GraphEdge[]
}

export interface ChatMessage {
  id: string
  role: 'user' | 'assistant'
  content: string
  citations?: Citation[]
  timestamp: number
}

export interface MeetingJobItem {
  id: string
  type: string
  status: string
  progress: number
  message: string
}

export interface MeetingJobsResponse {
  meeting_id: string
  jobs: MeetingJobItem[]
}
